wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubURLParser.java

```diff
@@ -97,6 +112,25 @@ public GithubUrl parse(String url) {
       }
     }
 
+    if (branchName == null) {
+      try {
+        HttpJsonResponse response =
+            requestFactory
+                .fromUrl("https://api.github.com/repos/" + repoUser + "/" + repoName)
```

Comment:
Would it make sense to create some singleton bean for requesting github API? So instead of calling http and handling requests manually here (which does not fit much to the scope of the class), we would just call a single method.